src/lib/arch/riscv/Instruction_execute.cc

```diff
@@ -827,6 +841,642 @@ void Instruction::execute() {
       break;
     }
 
+      // Control and Status Register extension (Zicsr)
+      // Currently do not read-modify-write ATOMICALLY
+      // Left unimplemented due to Capstone being unable to disassemble CSR
+      // addresses
+    case Opcode::RISCV_CSRRW: {  // CSRRW rd,csr,rs1
+      // Do nothing to allow progression
+
+      if (metadata.operands[1].reg == 2) {
+        // TODO this won't work properly in OoO core as rounding mode could be
+        // updated before instructions earlier in program order execute causing
+        // them to be rounded incorrectly. Rounding mode needs to be set on
+        // commit and all subsequent instructions must be performed with updated
+        // mode. e.g. this should be performed atomically
+
+        switch (operands[0].get<uint64_t>()) {
+          case 0:
+            fesetround(FE_TONEAREST);
+            break;
+          case 1:
+            fesetround(FE_TOWARDZERO);
+            break;
+          case 2:
+            fesetround(FE_DOWNWARD);
+            break;
+          case 3:
+            fesetround(FE_UPWARD);
+            break;
+          case 4:
+            fesetround(FE_TONEAREST);
+            break;
+          default:
+            // Invalid Case
+            // TODO any subsequent attempt to execute a floating-point operation
+            // with a dynamic rounding mode will raise an illegal instruction
+            // exception.
+            std::cerr << "[SimEng:RISC-V:Execute] Invalid rounding mode"
```

Comment:
Update prefix to be inline with the rest in the project i.e. `[SimEng:Instruction_execute]`. PLease apply throughout

Reply:
There is an argument to be made to keep ```:RISC-V:``` in some form at least to distinguish which instruction_execute is being referred to. Thoughts?

Reply:
In my opinion, given the user should know the ISA their binary targets it should be implicit which instruction_execute is referred to, plus it decreases the length of the prefix.

Interested to hear others thoughts on this - I'm happy to include ISA specific prefixes where necessary if thats the consensus @jj16791 @rahahahat @ABenC377 @JosephMoore25

Reply:
I'd vote to keep it without the ISA in the prefix as it's implied by the config file and I see it as redundant information for the user. 

However, as an aside, if we do switch to this sort of inclusion then it would also make sense to add a similar thing for core types, all files inside the arch-specific folders (e.g. `ExceptionHandler`), etc.

Reply:
@rahahahat @ABenC377 @JosephMoore25

Reply:
ISA printed in SimEng metadata so no need  to be within individual error messages